test/libyul/yulOptimizerTests/fullSuite/medium.yul

```diff
@@ -0,0 +1,28 @@
+{
+    function allocate(size) -> p {
+        p := mload(0x40)
+        mstore(0x40, add(p, size))
+    }
+    function array_index_access(array, index) -> p {
+        p := add(array, mul(index, 0x20))
+    }
+    pop(allocate(0x20))
+    let x := allocate(0x40)
+    mstore(array_index_access(x, 3), 2)
+}
+// ----
+// fullSuite
+// {
+//     {
+//         let _1 := 0x20
+//         let allocate_p
+//         allocate_p := mload(_3)
```

Comment:
Is the order wrong here? `_3` is only assigned later

Reply:
Yes, I think this is caused by the bug in the CSE. Will rebase.

Reply:
Actually yeah, this test is what caused me to investigate :)

Reply:
Updated.